src/box/lua/net_box.c

```diff
@@ -1155,7 +1156,8 @@ netbox_transport_connect(struct netbox_transport *transport)
 io_error:
 	assert(!diag_is_empty(diag_get()));
 	e = diag_last_error(diag_get());
-	box_error_raise(ER_NO_CONNECTION, "%s", e->errmsg);
+	diag_add(ClientError, ER_NO_CONNECTION);
+	error_vformat_msg(diag_last_error(diag_get()), e->errmsg, ap);
```

Comment:
Nit: it seems it's better to use `error_format_msg` here, you won't need `ap` then